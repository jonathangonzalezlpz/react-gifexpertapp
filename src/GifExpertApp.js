import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    //const categorias = ['OnePunch','Samurai X','Dragon Ball'];
    //const [categorias, setCategorias] = useState(['OnePunch']);
    const [categorias, setCategorias] = useState( defaultCategories );

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias={ setCategorias }/>
            <hr/>

            { 
                categorias.map( c => <GifGrid key={ c } 
                                            categoria={ c } /> )
            }

        </>
    )
};

export default GifExpertApp; 