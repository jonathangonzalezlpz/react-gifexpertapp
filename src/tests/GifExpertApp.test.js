import React from 'react';

import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en el componente <GifExpertApp />', () => {
    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categorias = ['One Punch', 'Dragon Ball Z'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    
    });
    
});
