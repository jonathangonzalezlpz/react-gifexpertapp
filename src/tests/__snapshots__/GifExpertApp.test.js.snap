// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    categoria="One Punch"
    key="One Punch"
  />
  <GifGrid
    categoria="Dragon Ball Z"
    key="Dragon Ball Z"
  />
</Fragment>
`;
